### ISSUE TITLE
error when adding subgroup in lefse

### ISSUE THREAD

**Issue by SonWende** [yiluheihei/microbiomeMarker#55]
Hey, 

thank you very much for this nice package that connects from phyloseq to lefse and other downstream methods, it's really a great help

I' m new to lefse and not sure yet which groups to compare. So i am trying out a few different runs.  
But when i attempt to add one metadata column as subgroup i get the following error: (note i am using a R jupyter notebook, not sure if this formats the error messages differently)

```
Error in if ((sx < sy) != dir_cmp || sx == sy) {: missing value where TRUE/FALSE needed
Traceback:
1. run_lefse(ps, taxa_rank = "Genus", wilcoxon_cutoff = 0.01, group = "AbioticStress", 
 .     subgroup = "Trichoderma", multigrp_strat = TRUE, kw_cutoff = 0.01, 
 .     norm = "CPM", lda_cutoff = 2)
2. purrr::map2_lgl(sig_otus, features_nms, ~test_rep_wilcoxon(subgroup, 
 .     grp_hie, .x, .y, wilcoxon_cutoff = wilcoxon_cutoff, multicls_strat = multigrp_strat, 
 .     strict = strict, sample_min = sample_min, only_same_subcls = only_same_subgrp, 
 .     curv = curv))
3. .f(.x[[i]], .y[[i]], ...)
4. test_rep_wilcoxon(subgroup, grp_hie, .x, .y, wilcoxon_cutoff = wilcoxon_cutoff, 
 .     multicls_strat = multigrp_strat, strict = strict, sample_min = sample_min, 
 .     only_same_subcls = only_same_subgrp, curv = curv)
```

Here is the sample code, the example phyloseq object is in the zip file.
The first commands with only the group works, the second command with added subgroup produces the error

```
library(microbiomeMarker)
ps <- readRDS("example.rds")

# check sample data
sample_data(ps)

# run lefse
# this works
run_lefse(ps,
          taxa_rank = "Genus",
          wilcoxon_cutoff = 0.01,
          group = "AbioticStress", 
          multigrp_strat = TRUE, # perform one-aganst-one compairsons
          kw_cutoff = 0.01,
          norm = "CPM",
          lda_cutoff = 2)

# this on does not work and produces the error
run_lefse(ps,
          taxa_rank = "Genus",
          wilcoxon_cutoff = 0.01,
          group = "AbioticStress", 
          subgroup="Trichoderma",
          multigrp_strat = TRUE, # perform one-aganst-one compairsons
          kw_cutoff = 0.01,
          norm = "CPM",
          lda_cutoff = 2)
```

When i run this comparison with group and subgroup in the conda version of lefse, it produces a result. The commands for that and the input table are also on the zip file

sessionInfo() is also in the zip file

[example_data.zip](https://github.com/yiluheihei/microbiomeMarker/files/8185706/example_data.zip)

Thank you very much for your help

Best Wishes, 
Sonja

**Comment by yiluheihei** [comment]
#62

**Comment by yiluheihei** [comment]
duplicated with #62

**Comment by yiluheihei** [comment]
fixed [02bbf87](https://github.com/yiluheihei/microbiomeMarker/commit/02bbf875077bd713aa921e4f941b06b719f41a37)
